src/arena.rs

```diff
@@ -656,12 +662,16 @@ impl ArenaAllocated for IndexPtr {
     }
 }
 
+#[repr(C)]
```

Comment:
This is crucial. Otherwise in my i686 test, header's address will be the same as slab's :-S